lighthouse-plugin-publisher-ads/test/smoke/run-smoke.js

```diff
@@ -0,0 +1,219 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+'use strict';
+
+/* eslint-disable no-console */
+const {promisify} = require('util'); // eslint-disable-line
+const execAsync = promisify(require('child_process').exec);
+const log = require('lighthouse-logger');
+const StaticServer = require('static-server');
+const {execSync} = require('child_process');
+
+const SMOKEHOUSE_PATH = require.resolve('lighthouse/lighthouse-cli/test/smokehouse/smokehouse.js');
+const CLI_ROOT = SMOKEHOUSE_PATH.replace('/test/smokehouse/smokehouse.js', '');
+const TMP_DIR = `${CLI_ROOT}/test/smokehouse/.tmp`;
+const CORE_DFNS_PATH = `${CLI_ROOT}/test/smokehouse/smoke-test-dfns.js`;
+const CUSTOM_DFNS_PATH = require.resolve('./smoke-test-dfns.js');
+
+
+const server = new StaticServer({
+  rootPath: `${__dirname}/fixtures`,
+  port: 8081,
+});
+
+/**
+ * Temporarily move files into core Lighthouse to allow for smokehouse usage.
+ */
+function init() {
+  const cmd = [
+    // Backup core dfns file.
+    `mv ${CORE_DFNS_PATH} ${CORE_DFNS_PATH}.backup`,
+    // Symlink custom dfns file.
+    `ln -sf ${CUSTOM_DFNS_PATH} ${CORE_DFNS_PATH}`,
+    // Create and symlink local `.tmp` directory.
+    `mkdir -p .tmp`,
+    `ln -sf ./.tmp ${TMP_DIR}`,
+    // Symlink `lighthouse-cli` in root.
+    `ln -sf ${CLI_ROOT} .`,
+  ];
+  execSync(cmd.join(' && '));
+}
+
+/**
+ * Revert changes from init().
+ */
+function cleanUp() {
+  const cmd = [
+    // Remove symlinked dfns file.
+    `rm ${CORE_DFNS_PATH}`,
+    // Restore backup dfns file.
+    `mv ${CORE_DFNS_PATH}.backup ${CORE_DFNS_PATH}`,
+    // Remove local and core `.tmp` dirs.
+    `rm -r .tmp`,
+    `rm ${TMP_DIR}`,
+    // Remove symlinked `lighthouse-cli`.
+    `rm lighthouse-cli`,
+  ];
+  execSync(cmd.join(' && '));
+}
+
```

Comment:
Given that this file is forked...lemme know if I need to read anything below here?